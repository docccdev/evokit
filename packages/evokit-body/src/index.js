import { createBlock } from 'evokit';

export const Body = createBlock('body', 'body', [
    'background',
    'color',
    'display',
    'indent',
    'size',
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
]);

export const BodySection = createBlock('div', 'body__section', ['display']);

Body.Section = BodySection;
