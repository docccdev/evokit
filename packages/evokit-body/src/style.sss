@import './mixins.sss'

@custom-media --ek-body-media-small only screen and (min-width: 480px)
@custom-media --ek-body-media-medium only screen and (min-width: 768px)
@custom-media --ek-body-media-large only screen and (min-width: 960px)
@custom-media --ek-body-media-wide only screen and (min-width: 1200px)
@custom-media --ek-body-media-huge only screen and (min-width: 1400px)

:root
    // base
    --ek-body-font-family: system-ui, -apple-system, Segoe UI, Roboto, Ubuntu, Cantarell, Noto Sans, sans-serif
    --ek-body-font-size: 13px
    --ek-body-font-weight: 400
    --ek-body-line-height: 1em
    --ek-body-min-width: 320px
    --ek-body-max-width: 1240px
    // prop 'body-indent'
    --ek-body-indent-xxs: 5px
    --ek-body-indent-xs: 10px
    --ek-body-indent-s: 15px
    --ek-body-indent-m: 20px
    --ek-body-indent-l: 25px
    --ek-body-indent-xl: 30px
    --ek-body-indent-xxl: 35px
    --ek-body-indent-3xl: 40px
    --ek-body-indent-4xl: 45px
    --ek-body-indent-5xl: 50px
    // prop 'body-width', 'body-width-min', 'body-width-max'
    --ek-body-width-xxs: 320px
    --ek-body-width-xs: 480px
    --ek-body-width-s: 768px
    --ek-body-width-m: 900px
    --ek-body-width-l: 1024px
    --ek-body-width-xl: 1280px
    --ek-body-width-xxl: 1366px
    --ek-body-width-3xl: 1440px
    --ek-body-width-4xl: 1600px
    --ek-body-width-5xl: 1920px
    // DEPRECATED prop 'body-size'
    --ek-body-max-width-xl: 1440px
    --ek-body-max-width-xxl: 1640px
    --ek-body-max-width-xxxl: 1840px

.body
    display: block
    font-family: var(--ek-body-font-family)
    font-size: var(--ek-body-font-size)
    line-height: var(--ek-body-line-height)
    font-weight: var(--ek-body-font-weight)
    min-width: var(--ek-body-min-width)
    margin: 0
    text-size-adjust: 100%
    -webkit-font-smoothing: subpixel-antialiased

.body__section
    display: block
    box-sizing: border-box
    margin-left: auto
    margin-right: auto
    min-width: var(--ek-body-min-width)
    max-width: var(--ek-body-max-width)

/**
 * Body display:
 *
 *  - body_display_block
 *  - body_display_none
 *
 **/

@each $value in (block, none)
    @mixin media .body_display_$(value)
        display: $value

/**
 * body-size:
 *
 *  - body_size_[xl|xxl|xxxl]
 *
 **/

@each $value in (xl, xxl, xxxl)
    @mixin media .body_size_$(value)
        .body__section
            max-width: var(--ek-body-max-width-$(value))

/**
 * body-width:
 *
 *  - body_width_none
 *  - body_width_xxs
 *  - body_width_xs
 *  - body_width_s
 *  - body_width_m
 *  - body_width_l
 *  - body_width_xl
 *  - body_width_xxl
 *  - body_width_3xl
 *  - body_width_4xl
 *  - body_width_5xl
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, 3xl, 4xl, 5xl)
    @mixin media .body_width_$(value)
        @if $value == none
            min-width: 0
        @else
            min-width: var(--ek-body-width-$(value))

        .body__section
            @if $value == none
                min-width: 0
                max-width: none
            @else
                min-width: var(--ek-body-width-$(value))
                max-width: var(--ek-body-width-$(value))

    @mixin media .body_width-min_$(value)
        @if $value == none
            min-width: 0
        @else
            min-width: var(--ek-body-width-$(value))

        .body__section
            @if $value == none
                min-width: 0
            @else
                min-width: var(--ek-body-width-$(value))

    @mixin media .body_width-max_$(value)
        .body__section
            @if $value == none
                max-width: none
            @else
                max-width: var(--ek-body-width-$(value))

/**
 * body-indent:
 *
 *  - body_indent_none
 *  - body_indent_xxs
 *  - body_indent_xs
 *  - body_indent_s
 *  - body_indent_m
 *  - body_indent_l
 *  - body_indent_xl
 *  - body_indent_xxl
 *  - body_indent_3xl
 *  - body_indent_4xl
 *  - body_indent_5xl
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, 3xl, 4xl, 5xl)
    @mixin media .body_indent_$(value)
        .body__section
            @if $value == none
                padding-left: 0
                padding-right: 0
            @else
                padding-left: var(--ek-body-indent-$(value))
                padding-right: var(--ek-body-indent-$(value))

/**
 * Overflow:
 *
 *  - body_overflow_*
 *  - body_overflow-x_*
 *  - body_overflow-y_*
 *
 **/

@each $value in (auto, hidden, scroll, visible)
    @mixin media .body_overflow_$(value)
        overflow: $value

    @mixin media .body_overflow-x_$(value)
        overflow-x: $value

    @mixin media .body_overflow-y_$(value)
        overflow-y: $value

/**
 * body-overflow-anchor:
 *
 *  - body_overflow-anchor_none
 *  - body_overflow-anchor_auto
 *
 **/

@each $value in (none, auto)
    @mixin media .body_overflow-anchor_$(value)
        overflow-anchor: $value

/**
 * Section display:
 *
 *  - body__section_display_block
 *  - body__section_display_none
 *
 **/

@each $value in (block, none)
    @mixin media .body__section_display_$(value)
        display: $value

/**
 * Section height:
 *
 *  - body__section_height_auto
 *  - body__section_height_inherit
 *  - body__section_height_1-1
 *
 **/

@each $value in (auto, inherit, 1-1)
    @mixin media .body__section_height_$(value)
        @if $value == 1-1
            height: 100%
        @else
            height: $value
