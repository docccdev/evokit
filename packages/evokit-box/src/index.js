// 3.2.0-beta.1
import { createBlock } from 'evokit';

export const Box = createBlock('div', 'box', [
    'align',
    'background',
    'display',
    'height',
    'opacity',
    'width',
    ['position', [
        'position',
        'place',
        'zindex',
    ]],
    ['border', [
        'border-color',
        'border-width',
        'border-style',
    ]],
    ['border-top', [
        'border-color-top',
        'border-width-top',
        'border-style-top',
    ]],
    ['border-right', [
        'border-color-right',
        'border-width-right',
        'border-style-right',
    ]],
    ['border-bottom', [
        'border-color-bottom',
        'border-width-bottom',
        'border-style-bottom',
    ]],
    ['border-left', [
        'border-color-left',
        'border-width-left',
        'border-style-left',
    ]],
    ['border-color', [
        ['border-color-top', 'border-color-bottom'],
        ['border-color-right', 'border-color-left'],
        'border-color-bottom',
        'border-color-left',
    ]],
    ['border-width', [
        ['border-width-top', 'border-width-bottom'],
        ['border-width-right', 'border-width-left'],
        'border-width-bottom',
        'border-width-left',
    ]],
    ['border-style', [
        ['border-style-top', 'border-style-bottom'],
        ['border-style-right', 'border-style-left'],
        'border-style-bottom',
        'border-style-left',
    ]],
    ['margin', [
        ['margin-top', 'margin-bottom'],
        ['margin-right', 'margin-left'],
        'margin-bottom',
        'margin-left',
    ]],
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'margin-lr',
    'margin-tb',
    'padding-lr',
    'padding-tb',
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
    'border-lr',
    'border-tb',
]);
