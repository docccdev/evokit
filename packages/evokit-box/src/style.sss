@import './mixins.sss'

@custom-media --ek-box-media-small only screen and (min-width: 480px)
@custom-media --ek-box-media-medium only screen and (min-width: 768px)
@custom-media --ek-box-media-large only screen and (min-width: 960px)
@custom-media --ek-box-media-wide only screen and (min-width: 1200px)
@custom-media --ek-box-media-huge only screen and (min-width: 1400px)

:root
    // prop 'box-padding', 'box-margin'
    --ek-box-indent-xxs: 5px
    --ek-box-indent-xs: 10px
    --ek-box-indent-s: 15px
    --ek-box-indent-m: 20px
    --ek-box-indent-l: 25px
    --ek-box-indent-xl: 30px
    --ek-box-indent-xxl: 35px
    // prop 'box-round'
    --ek-box-round-xxs: 2px
    --ek-box-round-xs: 4px
    --ek-box-round-s: 6px
    --ek-box-round-m: 8px
    --ek-box-round-l: 10px
    --ek-box-round-xl: 12px
    --ek-box-round-xxl: 14px
    // prop 'box-border-width'
    --ek-box-border-width-xxs: 1px
    --ek-box-border-width-xs: 2px
    --ek-box-border-width-s: 3px
    --ek-box-border-width-m: 4px
    --ek-box-border-width-l: 5px
    --ek-box-border-width-xl: 6px
    --ek-box-border-width-xxl: 7px

.box
    display: block
    box-sizing: border-box
    border-width: 0
    border-style: solid

    &:after
        clear: both

    &:after, &:before
        content: ''
        display: table

/**
 * Align:
 *
 *  - box_align_none
 *  - box_align_left
 *  - box_align_center
 *  - box_align_right
 *
 **/

@each $value in (none, left, center, right)
    @mixin media .box_align_$(value)
        @if $value == center
            float: none
            display: inline-block
            position: relative
            left: 50%
            transform: translateX(-50%)
        @else
            float: $value
            display: block
            position: static
            left: 0
            transform: translateX(0)

/**
 * Display:
 *
 *  - box_display_*
 *
 **/

@each $value in (none, inline, inline-block, block)
    @mixin media .box_display_$(value)
        display: $value

        @if $value == inline
            &:after, &:before
                content: none

/**
 * Position:
 *
 *  - box_position_*
 *
 **/

@each $value in (static, relative, absolute, fixed)
    @mixin media .box_position_$(value)
        position: $value

/**
 * Place:
 *
 *  - box_place_*
 *
 **/

@each $value in (top, top-left, top-right, center, center-left, center-right, bottom, bottom-left, bottom-right)
    @mixin media .box_place_$(value)
        @if $value == top
            top: 0
            left: 50%
            transform: translateX(-50%)
        @else if $value == top-left
            top: 0
            left: 0
            transform: none
        @else if $value == top-right
            top: 0
            right: 0
            transform: none
        @else if $value == center
            top: 50%
            left: 50%
            transform: translate(-50%, -50%)
        @else if $value == center-left
            top: 50%
            left: 0
            transform: translateY(-50%)
        @else if $value == center-right
            top: 50%
            right: 0
            transform: translateY(-50%)
        @else if $value == bottom
            bottom: 0
            left: 50%
            transform: translateX(-50%)
        @else if $value == bottom-left
            bottom: 0
            left: 0
            transform: none
        @else if $value == bottom-right
            bottom: 0
            right: 0
            transform: none

/**
 * Margin:
 *
 *  - box_margin_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, auto)
    @mixin media .box_margin_$(value)
        @if $value == none
            margin: 0
        @else if $value == auto
            margin: auto
        @else
            margin: var(--ek-box-indent-$(value))

    @mixin media .box_margin-tb_$(value)
        @if $value == none
            margin-top: 0
            margin-bottom: 0
        @else if $value == auto
            margin-top: auto
            margin-bottom: auto
        @else
            margin-top: var(--ek-box-indent-$(value))
            margin-bottom: var(--ek-box-indent-$(value))

    @mixin media .box_margin-lr_$(value)
        @if $value == none
            margin-left: 0
            margin-right: 0
        @else if $value == auto
            margin-left: auto
            margin-right: auto
        @else
            margin-left: var(--ek-box-indent-$(value))
            margin-right: var(--ek-box-indent-$(value))

    @mixin media .box_margin-left_$(value)
        @if $value == none
            margin-left: 0
        @else if $value == auto
            margin-left: auto
        @else
            margin-left: var(--ek-box-indent-$(value))

    @mixin media .box_margin-top_$(value)
        @if $value == none
            margin-top: 0
        @else if $value == auto
            margin-top: auto
        @else
            margin-top: var(--ek-box-indent-$(value))

    @mixin media .box_margin-right_$(value)
        @if $value == none
            margin-right: 0
        @else if $value == auto
            margin-right: auto
        @else
            margin-right: var(--ek-box-indent-$(value))

    @mixin media .box_margin-bottom_$(value)
        @if $value == none
            margin-bottom: 0
        @else if $value == auto
            margin-bottom: auto
        @else
            margin-bottom: var(--ek-box-indent-$(value))

/**
 * Padding:
 *
 *  - box_padding_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl)
    @mixin media .box_padding_$(value)
        @if $value == none
            padding: 0
        @else
            padding: var(--ek-box-indent-$(value))

    @mixin media .box_padding-tb_$(value)
        @if $value == none
            padding-top: 0
            padding-bottom: 0
        @else
            padding-top: var(--ek-box-indent-$(value))
            padding-bottom: var(--ek-box-indent-$(value))

    @mixin media .box_padding-lr_$(value)
        @if $value == none
            padding-left: 0
            padding-right: 0
        @else
            padding-left: var(--ek-box-indent-$(value))
            padding-right: var(--ek-box-indent-$(value))

    @mixin media .box_padding-left_$(value)
        @if $value == none
            padding-left: 0
        @else
            padding-left: var(--ek-box-indent-$(value))

    @mixin media .box_padding-top_$(value)
        @if $value == none
            padding-top: 0
        @else
            padding-top: var(--ek-box-indent-$(value))

    @mixin media .box_padding-right_$(value)
        @if $value == none
            padding-right: 0
        @else
            padding-right: var(--ek-box-indent-$(value))

    @mixin media .box_padding-bottom_$(value)
        @if $value == none
            padding-bottom: 0
        @else
            padding-bottom: var(--ek-box-indent-$(value))

/**
 * Round:
 *
 *  - box_round_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, full)
    @mixin media .box_round_$(value)
        @if $value == none
            border-radius: 0
        @else if $value == full
            border-radius: 50%
        @else
            border-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-top_$(value)
        @if $value == none
            border-top-left-radius: 0
            border-top-right-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
            border-top-right-radius: 50%
        @else
            border-top-left-radius: var(--ek-box-round-$(value))
            border-top-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-right_$(value)
        @if $value == none
            border-top-right-radius: 0
            border-bottom-right-radius: 0
        @else if $value == full
            border-top-right-radius: 50%
            border-bottom-right-radius: 50%
        @else
            border-top-right-radius: var(--ek-box-round-$(value))
            border-bottom-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-bottom_$(value)
        @if $value == none
            border-bottom-left-radius: 0
            border-bottom-right-radius: 0
        @else if $value == full
            border-bottom-left-radius: 50%
            border-bottom-right-radius: 50%
        @else
            border-bottom-left-radius: var(--ek-box-round-$(value))
            border-bottom-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-left_$(value)
        @if $value == none
            border-top-left-radius: 0
            border-bottom-left-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
            border-bottom-left-radius: 50%
        @else
            border-top-left-radius: var(--ek-box-round-$(value))
            border-bottom-left-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-top-left_$(value)
        @if $value == none
            border-top-left-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
        @else
            border-top-left-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-top-right_$(value)
        @if $value == none
            border-top-right-radius: 0
        @else if $value == full
            border-top-right-radius: 50%
        @else
            border-top-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-bottom-right_$(value)
        @if $value == none
            border-bottom-right-radius: 0
        @else if $value == full
            border-bottom-right-radius: 50%
        @else
            border-bottom-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-bottom-left_$(value)
        @if $value == none
            border-bottom-left-radius: 0
        @else if $value == full
            border-bottom-left-radius: 50%
        @else
            border-bottom-left-radius: var(--ek-box-round-$(value))


/**
 * Width:
 *
 *  - box_width_*
 *
 **/

@each $to in (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
    @each $from in (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
        @if $from < $to or ($from == 1 and $to == 1)
            @mixin media .box_width_$(from)-$(to)
                width: calc($from / $to * 100%)

@mixin media .box_width_auto
    width: auto

/**
 * Height:
 *
 *  - box_height_*
 *
 **/

@mixin media .box_height_1-1
    height: 100%

@mixin media .box_height_auto
    height: auto

/**
 * Overflow:
 *
 *  - box_overflow_*
 *  - box_overflow-x_*
 *  - box_overflow-y_*
 *
 **/

@each $value in (auto, hidden, scroll, visible)
    @mixin media .box_overflow_$(value)
        overflow: $value

    @mixin media .box_overflow-x_$(value)
        overflow-x: $value

    @mixin media .box_overflow-y_$(value)
        overflow-y: $value
