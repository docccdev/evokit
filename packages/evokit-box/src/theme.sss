@import './mixins.sss'

:root
    --ek-box-background-TEMPLATE_NAME: var(--background)
    --ek-box-border-TEMPLATE_NAME: var(--border)

@mixin media .box_background_TEMPLATE_NAME
    background: var(--ek-box-background-TEMPLATE_NAME)

/* BORDER */

@mixin media .box_border_TEMPLATE_NAME
    border: 1px solid var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-top_TEMPLATE_NAME
    border-top: 1px solid var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-right_TEMPLATE_NAME
    border-right: 1px solid var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-bottom_TEMPLATE_NAME
    border-bottom: 1px solid var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-left_TEMPLATE_NAME
    border-left: 1px solid var(--ek-box-border-TEMPLATE_NAME)

/* BORDER COLOR */

@mixin media .box_border-color_TEMPLATE_NAME
    border-color: var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-color-top_TEMPLATE_NAME
    border-top-color: var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-color-right_TEMPLATE_NAME
    border-right-color: var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-color-bottom_TEMPLATE_NAME
    border-bottom-color: var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-color-left_TEMPLATE_NAME
    border-left-color: var(--ek-box-border-TEMPLATE_NAME)

/* DEPRECATED */

@mixin media .box_border-tb_TEMPLATE_NAME
    border-top: 1px solid var(--ek-box-border-TEMPLATE_NAME)
    border-bottom: 1px solid var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-lr_TEMPLATE_NAME
    border-left: 1px solid var(--ek-box-border-TEMPLATE_NAME)
    border-right: 1px solid var(--ek-box-border-TEMPLATE_NAME)
