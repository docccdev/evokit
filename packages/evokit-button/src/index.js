import { createBlock } from 'evokit';

export const Button = createBlock('button', 'button', [
    'display',
    'events',
    'height',
    'size',
    'theme',
    'valign',
    'weight',
    'width',
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
]);
