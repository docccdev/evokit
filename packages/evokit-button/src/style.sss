@import './mixins.sss'

@custom-media --ek-button-media-small only screen and (min-width: 480px)
@custom-media --ek-button-media-medium only screen and (min-width: 768px)
@custom-media --ek-button-media-large only screen and (min-width: 960px)
@custom-media --ek-button-media-wide only screen and (min-width: 1200px)
@custom-media --ek-button-media-huge only screen and (min-width: 1400px)

:root
    // base
    --ek-button-border-radius: 2px
    --ek-button-border-width: 1px
    --ek-button-font-size: inherit
    --ek-button-font-weight: bold
    --ek-button-line-height: 1em
    --ek-button-padding: 10px 15px
    --ek-button-transition: 150ms ease-in-out
    // prop 'button-padding'
    --ek-button-indent-xxs: 5px
    --ek-button-indent-xs: 10px
    --ek-button-indent-s: 15px
    --ek-button-indent-m: 20px
    --ek-button-indent-l: 25px
    --ek-button-indent-xl: 30px
    --ek-button-indent-xxl: 35px
    --ek-button-indent-3xl: 40px
    --ek-button-indent-4xl: 45px
    --ek-button-indent-5xl: 50px
    // prop 'button-round'
    --ek-button-round-xxs: 2px
    --ek-button-round-xs: 4px
    --ek-button-round-s: 6px
    --ek-button-round-m: 8px
    --ek-button-round-l: 10px
    --ek-button-round-xl: 12px
    --ek-button-round-xxl: 14px
    --ek-button-round-3xl: 16px
    --ek-button-round-4xl: 18px
    --ek-button-round-5xl: 20px
    // prop 'button-size'
    --ek-button-size-small: 11px
    --ek-button-size-default: var(--ek-button-font-size)
    --ek-button-size-big: 15px
    --ek-button-size-h1: 24px
    --ek-button-size-h2: 22px
    --ek-button-size-h3: 20px
    --ek-button-size-h4: 18px
    --ek-button-size-h5: 16px
    --ek-button-size-h6: 14px

.button
    position: relative
    display: inline-flex
    align-items: center
    justify-content: center
    font: inherit
    color: inherit
    box-sizing: border-box
    appearance: none
    text-decoration: none
    background: transparent
    user-select: none
    outline: none
    vertical-align: middle
    cursor: pointer
    transition-property: color, background-color, border-color
    max-width: 100%
    border: var(--ek-button-border-width) solid transparent
    padding: var(--ek-button-padding)
    border-radius: var(--ek-button-border-radius)
    font-size: var(--ek-button-font-size)
    font-weight: var(--ek-button-font-weight)
    line-height: var(--ek-button-line-height)
    transition: var(--ek-button-transition)

    &:disabled
        cursor: default
        pointer-events: none

    &::-moz-focus-inner
        border-style: none

/**
 * Display:
 *
 *  - button_display_block
 *  - button_display_none
 *
 **/

@each $value in (inline-flex, none)
    @mixin media .button_display_$(value)
        display: $value


/**
 * Height:
 *
 *  - button_height_auto
 *  - button_height_inherit
 *  - button_height_1-1
 *
 **/

@each $value in (auto, inherit, 1-1)
    @mixin media .button_height_$(value)
        @if $value == 1-1
            height: 100%
        @else
            height: $value


/**
 * Width:
 *
 *  - button_width_auto
 *  - button_width_inherit
 *  - button_width_1-1
 *
 **/

@each $value in (auto, inherit, 1-1)
    @mixin media .button_width_$(value)
        @if $value == 1-1
            width: 100%
        @else
            width: $value


/**
 * Padding:
 *
 *  - button_padding_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, 3xl, 4xl, 5xl)
    @mixin media .button_padding_$(value)
        @if $value == none
            padding: 0
        @else
            padding: var(--ek-button-indent-$(value))

    @mixin media .button_padding-left_$(value)
        @if $value == none
            padding-left: 0
        @else
            padding-left: var(--ek-button-indent-$(value))

    @mixin media .button_padding-top_$(value)
        @if $value == none
            padding-top: 0
        @else
            padding-top: var(--ek-button-indent-$(value))

    @mixin media .button_padding-right_$(value)
        @if $value == none
            padding-right: 0
        @else
            padding-right: var(--ek-button-indent-$(value))

    @mixin media .button_padding-bottom_$(value)
        @if $value == none
            padding-bottom: 0
        @else
            padding-bottom: var(--ek-button-indent-$(value))


/**
 * Round:
 *
 *  - button_round_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, 3xl, 4xl, 5xl, full)
    @mixin media .button_round_$(value)
        @if $value == none
            border-radius: 0
        @else if $value == full
            border-radius: 50%
        @else
            border-radius: var(--ek-box-round-$(value))

    @mixin media .button_round-top-left_$(value)
        @if $value == none
            border-top-left-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
        @else
            border-top-left-radius: var(--ek-box-round-$(value))

    @mixin media .button_round-top-right_$(value)
        @if $value == none
            border-top-right-radius: 0
        @else if $value == full
            border-top-right-radius: 50%
        @else
            border-top-right-radius: var(--ek-box-round-$(value))

    @mixin media .button_round-bottom-right_$(value)
        @if $value == none
            border-bottom-right-radius: 0
        @else if $value == full
            border-bottom-right-radius: 50%
        @else
            border-bottom-right-radius: var(--ek-box-round-$(value))

    @mixin media .button_round-bottom-left_$(value)
        @if $value == none
            border-bottom-left-radius: 0
        @else if $value == full
            border-bottom-left-radius: 50%
        @else
            border-bottom-left-radius: var(--ek-box-round-$(value))

/**
 * Font weight:
 *
 *  - button_weight_*
 *
 **/

@each $value in (thin, light, normal, medium, bold, black)
    @mixin media .button_weight_$(value)
        @if $value == thin
            font-weight: 100
        @else if $value == light
            font-weight: 300
        @else if $value == normal
            font-weight: 400
        @else if $value == medium
            font-weight: 500
        @else if $value == bold
            font-weight: 700
        @else if $value == black
            font-weight: 900

/**
 * Font size:
 *
 *  - button_size_*
 *
 **/

@each $value in (none, inherit, small, default, big, h1, h2, h3, h4, h5, h6)
    @mixin media .button_size_$(value)
        @if $value == none
            font-size: 0
        @else if $value == inherit
            font-size: inherit
        @else
            font-size: var(--ek-button-size-$(value))

/**
 * Valign:
 *
 *  - button_valign_*
 *
 **/

@each $value in (baseline, bottom, middle, sub, super, text-bottom, text-top, top)
    @mixin media .button_valign_$(value)
        vertical-align: $value

/**
 * Pointer events:
 *
 *  - button_events_*
 *
 **/

@each $value in (none, auto)
    @mixin media .button_events_$(value)
        pointer-events: $value
