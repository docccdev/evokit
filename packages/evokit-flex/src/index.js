import { createBlock } from 'evokit';

export const Flex = createBlock('div', 'flex', [
    'direction',
    'display',
    'items',
    'wrap',
    ['content', [
        'content-align',
        'content-justify',
    ]],
]);

export const FlexItem = createBlock('div', 'flex__item', [
    'align',
    'display',
    'order',
]);

Flex.Item = FlexItem;
