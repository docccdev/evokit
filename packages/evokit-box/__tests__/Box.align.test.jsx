import React from 'react';
import { shallow } from 'enzyme';
import barista from 'seed-barista';
import style from '../style.css';
import { Box } from '../src';

describe('<Box box-align="..." />', () => {
    it('props box-align', () => {
        const wrapper = shallow(<Box box-align='left' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<div class="ek-box ek-box_align_left"></div>');
    });

    it('css', () => {
        const css = barista({ content: style });

        expect(css.rule('.ek-box_align_none').exists()).toBeTruthy();
        expect(css.rule('.ek-box_align_left').exists()).toBeTruthy();
        expect(css.rule('.ek-box_align_center').exists()).toBeTruthy();
        expect(css.rule('.ek-box_align_right').exists()).toBeTruthy();
    });
});
