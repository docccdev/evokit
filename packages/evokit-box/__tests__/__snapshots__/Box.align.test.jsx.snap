// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box box-align="..." /> props box-align 1`] = `
<div
  className="ek-box ek-box_align_left"
/>
`;
